Lib/test/test_dataclasses/__init__.py

```diff
@@ -3367,6 +3367,24 @@ class A:
         self.assertFalse(hasattr(A, "__slots__"))
         self.assertTrue(hasattr(B, "__slots__"))
 
+    def test_slots_qualname(self):
+        # Test that __qualname__ is set correctly when using slots
+        @dataclass(slots=True)
+        class C:
+            x: int
+
+            def __init_subclass__(cls):
+                expected = f'TestSlots.test_slots_qualname.<locals>.{cls.__name__}'
+                self.assertEqual(cls.__qualname__, expected)
+
+        self.assertEqual(C.__qualname__, 'TestSlots.test_slots_qualname.<locals>.C')
+
+        @dataclass(slots=True)
+        class D(C):
+            pass
+
+        self.assertEqual(D.__qualname__, 'TestSlots.test_slots_qualname.<locals>.D')
```

Comment:
Can you please run your tests as `-m test test_dataclasses` and as `python.exe Lib/test/test_dataclasses/__init__.py` directly? qualname might be changed

Reply:
I assume @sobolevn thinks the imported class name might change.

Reply:
Yeah, I am not sure it will, just double checking.